File: documentation/manual/working/scalaGuide/main/http/code/scalaguide/http/routing/relative/controllers/Relative.scala
/*
 * Copyright (C) from 2022 The Play Framework Contributors <https://github.com/playframework>, 2011-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package scalaguide.http.routing.relative
//#relative-controller
package controllers

import javax.inject._

import play.api.mvc._

@Singleton
class Relative @Inject() (cc: ControllerComponents) extends AbstractController(cc) {
  def helloview: Action[AnyContent] = Action { implicit request => Ok(views.html.hello("Bob")) }

  def hello(name: String): Action[AnyContent] = Action {
    Ok(s"Hello $name!")
  }
}

File: documentation/manual/working/javaGuide/main/http/code/javaguide/http/routing/relative/controllers/Relative.java
/*
 * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
 */
// #relative-controller
// ###replace: package controllers;
package javaguide.http.routing.relative.controllers;

import play.*;
import play.mvc.*;

public class Relative extends Controller {

  public Result helloview() {
    // ###replace:         ok(views.html.hello.render("Bob", request()));
    return ok(javaguide.http.routing.relative.views.html.hello.render("Bob", request()));
  }

  public Result hello(String name) {
    return ok("Hello " + name + "!");
  }
}
